### PR TITLE
Extract indexApiVersion

### PR DESCRIPTION
**Pull request by emmberk**

### CODE REVIEW

File: src/main/java/com/connexta/store/config/IndexClientConfiguration.java [hunk 1]
```diff
@@ -27,9 +27,10 @@ public RestTemplate nonBufferingRestTemplate() {
   }
 
   @Bean
-  public IndexClient searchClient(
+  public IndexClient indexClient(
       @NotNull @Named("nonBufferingRestTemplate") final RestTemplate restTemplate,
-      @NotBlank @Value("${endpointUrl.index}") final String indexEndpoint) {
-    return new IndexClient(restTemplate, indexEndpoint);
+      @NotBlank @Value("${endpointUrl.index}") final String indexEndpoint,
```
Comment (figliold) [inline src/main/java/com/connexta/store/config/IndexClientConfiguration.java]: ❓ Can indexApiVersion be final too?

File: src/main/java/com/connexta/store/clients/IndexClient.java [hunk 2]
```diff
@@ -54,11 +51,10 @@ public String getFilename() {
           }
         });
     final HttpHeaders httpHeaders = new HttpHeaders();
-    // TODO inject IndexApi version
-    httpHeaders.set("Accept-Version", "0.1.0");
+    httpHeaders.set("Accept-Version", indexApiVersion);
```
Comment (figliold) [inline src/main/java/com/connexta/store/clients/IndexClient.java]: ❓ Do you think we should make Accept-Version a constant?

File: src/test/resources/application.yml [hunk 3]
```diff
@@ -3,5 +3,8 @@ server:
 endpointUrl:
   retrieve: http://localhost:${server.port}/mis/product/
   index: http://search:8080/index/
+endpoints:
+  index:
+    version: testIndexApiVersion
```
Comment (jhunzik) [inline src/test/resources/application.yml]: ❓ Is this supposed to be a placeholder `${testIndexApiVersion}` or an actual value for the version?

Comment (emmberk) [reply src/test/resources/application.yml]: This is the actual value.

**Comment by emmberk** [comment]
I heroed this locally

**Pull request merged**